### PR TITLE
Use a system unit to improve security

### PR DESCRIPTION
**Pull request by seb128**
I'm proposing that one to help moving things forward but it's untested, the systemd job seems to work as it's installed and systemctl status report it as started once it's dbus activated but I'm unsure how to drive the service. Robert, could you check if activating a subscription is still working with the security feature activated? If so we should probably be able to merge the change as it is now, we can always tweak later if we want more confinement.

### CODE REVIEW

File: src/meson.build [hunk 1]
```diff
@@ -19,9 +19,16 @@ gdbus_src = custom_target(
 
 daemon_builddir = meson.current_build_dir()
 
+unitdir = ''
+systemd = dependency('systemd', required: false)
+if systemd.found()
```
Comment (3v1n0) [inline src/meson.build:24]: Maybe warn about otherwise?

Comment (seb128) [reply src/meson.build:24]: I copied the section from bolt and I've no strong opinion, I'm fine warning or erroring out. Robert any preference?

Comment (robert-ancell) [reply src/meson.build:24]: I would make systemd a hard dependency - this package is only going to be used on Ubuntu and so we don't have to handle the case that it's not present.

File: src/meson.build [hunk 2]
```diff
@@ -41,6 +48,19 @@ service_file = configure_file(input: 'com.canonical.UbuntuAdvantage.service.in',
...
 install_data(service_file, install_dir: join_paths(dbus_datadir, 'system-services'))
 
+# Systemd unit
+systemd_unit = configure_file(
+  input: 'ubuntu-advantage-desktop-daemon.service.in',
+  output: 'ubuntu-advantage-desktop-daemon.service',
+  configuration: conf
+)
+
+if unitdir != ''
```
Comment (robert-ancell) [inline src/meson.build:58]: Drop the if (hard dependency).

File: src/meson.build [hunk 3]
```diff
@@ -19,9 +19,16 @@ gdbus_src = custom_target(
...
+unitdir = ''
+systemd = dependency('systemd', required: false)
+if systemd.found()
+  unitdir = systemd.get_pkgconfig_variable('systemdsystemunitdir')
+endif
+
 conf = configuration_data()
 conf.set_quoted('GETTEXT_PACKAGE', 'ubuntu-advantage-desktop-daemon')
 conf.set_quoted('PROJECT_VERSION', meson.project_version())
+conf.set('libexecdir', libexecdir)
```
Comment (robert-ancell) [inline src/meson.build:31]: I would make a separate conf object just for the `systemd_unit` - that way we can't accidentally make a collision with unexpected defines / substitutions.

**Review by robert-ancell — CHANGES_REQUESTED** [review]
Thanks Seb! Looks great, just the small changes I commented on.

File: src/meson.build (hunk 1 above)
Comment (seb128) [reply src/meson.build]: updated

File: src/meson.build [hunk 4]
```diff
@@ -19,6 +19,13 @@ gdbus_src = custom_target(
 
 daemon_builddir = meson.current_build_dir()
 
+unitdir = ''
```
Comment (robert-ancell) [inline src/meson.build:22]: This line does nothing now

File: src/meson.build [hunk 5]
```diff
@@ -19,6 +19,13 @@ gdbus_src = custom_target(
...
+systemd = dependency('systemd')
+unitdir = systemd.get_pkgconfig_variable('systemdsystemunitdir')
+
+confsystemd = configuration_data()
```
Comment (robert-ancell) [inline src/meson.build:26]: I'd move this to just above `systemd_unit = configure_file(`

File: src/meson.build (hunk 4 above)
Comment (seb128) [reply src/meson.build]: removed

File: src/meson.build (hunk 5 above)
Comment (seb128) [reply src/meson.build]: right, makes sense, I moved it to one section

**Comment by seb128** [comment]
Alright, I pushed some extra changes, the meson tweaks requested and I granted some extra permissions since the service needs to be able to contact the server and to write in /var. I confirmed that now status, attach and detach are working, it should be ready to merge from my perspective.

**Review by robert-ancell — APPROVED** [review]

**Comment by robert-ancell** [comment]
We're missing the build-dependency in `debian/control` but I'll add that as a separate commit.

**Pull request merged**